ql/src/java/org/apache/hadoop/hive/ql/lockmgr/DbTxnManager.java

```diff
@@ -485,6 +480,26 @@ private void clearLocksAndHB() {
     stopHeartbeat();
   }
 
+  private void cleanupDirForCTAS() {
```

Comment:
Do I understand correctly, that we do not create a table for CTAS, we just create a `destinationTable` object which behaves as a table, but it is not created in the HMS?
So basically we do not really have to drop the table, we just have to drop the directory?

Reply:
Yes you are correct.

Reply:
Creation of table happens here - As part of DDLTask. 
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/ddl/table/create/CreateTableOperation.java#L104-L117

The test case I have written is essentially manipulating the contents of CreateTableDesc to null so that creation of table fails. But in reality, if there is a failure before creation of table then the data written will stay and is not associated to any table and is never cleaned up.

Reply:
Thanks for the explanation!